pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,69 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authValidationTransport struct {
+	Wrapped              http.RoundTripper
+	AllowedGroups        []string
+	LdapGroupsHeaderName string
+}
+
+func Contains(s []string, target string) bool {
+	for _, c := range s {
+		if target == c {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *authValidationTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	incomingLdapGroups := req.Header[r.LdapGroupsHeaderName]
+	allowedList := r.AllowedGroups
+	for _, g := range incomingLdapGroups {
```

Comment:
Question:
Are we ok with iterating through every incoming ldap group and every allowed ldap group as arrays/slices? Hypothetically, the allowed ldap groups array should not be massive so this should work without too much performance tax.

However, I did contemplate making these lists of ldap groups maps, the problem is if I convert a list into a map, I'm still going to pay a large performance tax for that conversion unless we cache it somewhere. Or we could possibly store the values of the allowed ldap groups as maps in the api.yaml if that is permitted. The main appeal of a map versus an array is checking for membership is very fast but the size of this data may not benefit as much from such an improvement since the arrays should be small.

Reply:
I believe the performance hit of scanning the list will generally be negligible. I'm fine with keeping it as an array but I also won't complain if you convert things into a map in the constructor.

Reply:
FWIW, the operation being performed here is a set intersection. Sets in Go are most commonly handled using maps but there is no standard library set type. If you converted the original list of acceptable header values to something like a `map[string]bool` then you could simplify this logic to:

```golang
for _, value := range givenValues {
 if acceptableValues[value] {
    return r.Wrapped.RoundTrip(req)
  }
}
return newError(http.StatusUnauthorized, "...")
```